infra/modules/auth-github-actions/main.tf

```diff
@@ -20,6 +23,83 @@ resource "aws_iam_role_policy_attachment" "custom" {
   policy_arn = var.iam_role_policy_arns[count.index]
 }
 
+# Create a policy to push images to ECR and update ECS task definitions
+resource "aws_iam_policy" "deploy" {
+  name        = "${var.github_actions_role_name}-deploy"
+  description = "A policy for a machine user to deploy releases"
+  policy      = data.aws_iam_policy_document.deploy.json
+}
+
+resource "aws_iam_role_policy_attachment" "deploy" {
+  role       = aws_iam_role.github_actions.name
+  policy_arn = aws_iam_policy.deploy.arn
+}
+
+data "aws_iam_policy_document" "deploy" {
+  # Required to push a new docker image to ECR
+  statement {
+    sid    = "AccessECR"
+    effect = "Allow"
+    actions = [
+      "ecr:GetAuthorizationToken",
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "PublishImage"
+    effect = "Allow"
+    actions = [
+      "s3:ListBucket",
+      "s3:GetObject",
+      "dynamodb:GetItem",
+    ]
+    resources = [
+      "arn:aws:dynamodb:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:table/*",
+      "arn:aws:s3:::*",
+    ]
+  }
```

Comment:
this looks like it's the TFstate backend, not anything to do with ECR?

Reply:
Missing: 
```terraform
                "ecr:CompleteLayerUpload",
                "ecr:UploadLayerPart",
                "ecr:InitiateLayerUpload",
                "ecr:BatchCheckLayerAvailability",
                "ecr:PutImage"
            ],
```

Reply:
I'll give it a shot without these:
```
"s3:ListBucket",
"s3:GetObject",
"dynamodb:GetItem",
```
but it doesn't seem to need those other ones 🤔

Reply:
I tried it without these three, but because the CD files get variables from terraform, we run `terraform init` and need to be able to access the state: https://github.com/navapbc/wic-participant-recertification-portal/actions/runs/4441708285/jobs/7797114962